sklearn/utils/testing.py

```diff
@@ -697,5 +705,27 @@ def check_skip_travis():
     if os.environ.get('TRAVIS') == "true":
         raise SkipTest("This test needs to be skipped on Travis")
 
+class TempMemMap(object):
+    def __init__(self, X):
```

Comment:
please put `mmap_mode='r'` here and use to make it possible to use a different mode.